File: ui/react/src/action/AddTable.ts
import Table from '../model/Table';
import SheetAction from './SheetAction';
import Action from './Action';

class AddTable extends SheetAction {
    public table: Table;

    public static deserialize(input: any) {
        const sheetName: string = input.sheetName;
        const table: Table = Table.deserialize(input.table);
        return new AddTable(sheetName, table);
    }

    constructor(sheetName: string, table: Table) {
        super(sheetName);
        this.table = table;
    }

    public wrapper(): Action {
        const action = new Action();
        action.addTable = this;
        return action;
    }

}

export default AddTable;

File: core/src/main/java/com/ctrip/ferriswheel/core/action/AddTable.java
package com.ctrip.ferriswheel.core.action;

import com.ctrip.ferriswheel.common.action.Action;
import com.ctrip.ferriswheel.common.table.Table;

public final class AddTable extends SheetAction implements Action {
    private Table tableData;

    public AddTable() {
    }

    public AddTable(String sheetName, Table tableData) {
        super(sheetName);
        this.tableData = tableData;
    }

    @Override
    public String toString() {
        return "AddTable{" +
                "sheetName='" + getSheetName() + '\'' +
                ", tableData='" + tableData + '\'' +
                '}';
    }

    public Table getTableData() {
        return tableData;
    }

    public void setTableData(Table tableData) {
        this.tableData = tableData;
    }
}
